### PR TITLE
Additional typechecking with prettier update

### PR DESCRIPTION
**Pull request by bcolloran**
updates Prettier, eslint, and typescript to latest versions.

Note that you may need to tweak your vscode settings to get format-on-save working --

I needed to do this to get it working for svelte files:
"""ok, i think for me it was a bad vscode setting. I went to a svelte file, borked the formatting, then did ctrl+shift+p to "format doc" through the command bar. It told me that the prettier plugin didn't know how to format svelte files, and let me choose the svlete plugin. seems to be working now""" (see this slack conversation, which also includes discussion of what Brian needed to do: https://rilldata.slack.com/archives/C04AGHL77PS/p1704405722497439?thread_ts=1704400984.394149&cid=C04AGHL77PS)

Eric needed to do this:
"""To get format-on-save working for me in VSCode, I also had to add to my VSCode config "prettier.documentSelectors": ["**/*.svelte"] [described here](https://github.com/sveltejs/prettier-plugin-svelte?tab=readme-ov-file#usage-with-tailwind-prettier-plugin)."""

### CODE REVIEW

File: docs/package.json [hunk 1]
```diff
@@ -18,8 +18,8 @@
     "clsx": "^1.1.1",
     "docusaurus-gtm-plugin": "^0.0.2",
     "prism-react-renderer": "^1.3.3",
-    "react-dom": "^17.0.2",
     "react": "^17.0.2",
+    "react-dom": "^17.0.2",
```
Comment (bcolloran) [inline docs/package.json:22]: automatic resorting of deps, no actual change in /docs

File: .eslintignore [hunk 2]
```diff
@@ -1,2 +1,5 @@
 web-admin/build
-web-local/build
\ No newline at end of file
+web-local/build
+**/svelte.config.js
+
+web-common/src/components/modal/*.js
```
Comment (bcolloran) [inline .eslintignore:5]: ignored vendored files

File: .eslintrc.cjs [hunk 3]
```diff
@@ -3,7 +3,8 @@ module.exports = {
   extends: [
     "eslint:recommended",
     "plugin:@typescript-eslint/recommended",
-    'plugin:svelte/recommended',
+    "plugin:@typescript-eslint/recommended-type-checked",
```
Comment (bcolloran) [inline .eslintrc.cjs:6]: added current recommended eslint typechecking rules, and then disabled failing rules below for incremental cleanup

File: package.json [hunk 4]
```diff
@@ -21,5 +21,15 @@
...
+  },
+  "devDependencies": {
+    "@typescript-eslint/eslint-plugin": "^6.17.0",
+    "@typescript-eslint/parser": "^6.17.0",
+    "eslint": "^8.56.0",
+    "eslint-config-prettier": "^9.1.0",
+    "eslint-plugin-svelte": "^2.35.1",
+    "prettier": "^3.1.1",
+    "prettier-plugin-svelte": "^3.1.2",
+    "typescript": "^5.3.3"
```
Comment (bcolloran) [inline package.json:33]: moved code quality dev deps to root package.json

File: web-admin/package.json [hunk 5]
```diff
@@ -20,25 +20,17 @@
...
     "postcss": "^8.4.21",
-    "prettier": "^2.6.2",
-    "prettier-plugin-svelte": "^2.10.1",
     "svelte": "^4.0.0",
     "svelte-check": "^3.4.3",
     "svelte-popperjs": "^1.3.2",
     "svelte-preprocess": "^5.0.3",
     "tailwindcss": "^3.2.7",
-    "typescript": "^5.0.0",
     "web-common": "*"
```
Comment (bcolloran) [inline web-admin/package.json:33]: moved code quality dev deps to root package.json

File: .eslintignore (hunk 2 above)
Comment (ericpgreen2) [reply .eslintignore:5]: This comment would be good in the file itself :)

File: docs/package.json (hunk 1 above)
Comment (ericpgreen2) [reply docs/package.json:22]: I tried to replicate this re-sorting behavior, but I see `*.json` is included in the `.prettierignore` file. How did these re-sort automatically?

File: tsconfig.json [hunk 6]
```diff
@@ -43,5 +43,7 @@
     "**/*.d.ts",
     "**/*.ts",
     "**/*.svelte",
+    "prettier.config.js",
```
Comment (ericpgreen2) [inline tsconfig.json:46]: This file doesn't exist anymore

File: tsconfig.json [hunk 7]
```diff
@@ -43,5 +43,7 @@
...
+    "**/rollup.config.js",
```
Comment (ericpgreen2) [inline tsconfig.json:47]: Can we instead use `rollup.config.ts`?

File: web-auth/src/components/Auth.svelte [hunk 8]
```diff
@@ -120,14 +120,14 @@
...
                 ? err.description
                 : typeof err?.policy === "string"
-                ? err.policy
-                : typeof err?.error_description === "string"
-                ? err.error_description
-                : err?.message;
+                  ? err.policy
+                  : typeof err?.error_description === "string"
+                    ? err.error_description
+                    : err?.message;
```
Comment (ericpgreen2) [inline web-auth/src/components/Auth.svelte:126]: This is great!

**Review by ericpgreen2 — APPROVED** [review]
Nice cleanup 💪. Left a couple questions inline.

To get format-on-save working for me in VSCode, I also had to add to my VSCode config `"prettier.documentSelectors": ["**/*.svelte"]` [described here](https://github.com/sveltejs/prettier-plugin-svelte?tab=readme-ov-file#usage-with-tailwind-prettier-plugin). Maybe call that out for others in the PR description?

**Review by briangregoryholmes — APPROVED** [review]

File: docs/package.json (hunk 1 above)
Comment (bcolloran) [reply docs/package.json:22]: they actually re-sorted when I was `npm installing` stuff, not from prettier. <shrug>

File: tsconfig.json (hunk 7 above)
Comment (bcolloran) [reply tsconfig.json:47]: Hmm, I'm not sure. The rollup config we have is currently js not ts, and I don't know whether rollup includes a tsc preprocessing+compilation step for ts files. I think we can investigate in follow up work.

File: tsconfig.json (hunk 6 above)
Comment (bcolloran) [reply tsconfig.json:46]: removed -- nice catch, thanks!

**Review by djbarnwal — APPROVED** [review]

**Pull request merged**